### DISCUSSION TITLE
Sequential Jobs

### DISCUSSION THREAD

**Discussion by AleksandrZhukov** [immich-app/immich#14960] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

I'm interested in running jobs sequentially to better manage resource usage on my system (raspberry pi). 
Currently, it seems that all jobs (thumbnail generation, video transcoding, etc.) run concurrently. Is there a way to configure Immich to process jobs one after another? For example, completing thumbnail generation for all assets before starting video transcoding. This would help me avoid overloading my system. 
If this is not currently possible, are there any plans to implement a job queuing or prioritization system?

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by mertalev** [comment]
You can set their concurrency to 1 and pause certain queues if necessary (transcoding in particular is the most intensive, but it internally has a "niceness" setting so it lets other tasks take priority). There is no way to force only 1 job to be run at a time and this is unlikely to change in the foreseeable future.

**Reply by AleksandrZhukov** [reply]
Thanks, that's what I was doing manually, but thought maybe there is a way to achieve that another way

**Comment by quiode** [comment]
I just want to add that this would still be very useful for me. My graphics card can't handle multiple jobs at the same time but one after another works fine. It would be really nice if I could enable that.

**Reply by fmohiy-eg** [reply]
Same here. My GPU gets filled with one ML model and the other silently revert to CPU.
